src/main/java/org/opensearch/neuralsearch/processor/rerank/context/ContextSourceFetcher.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.rerank.context;
+
+import java.util.Map;
+
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.core.action.ActionListener;
+
+/**
+ * Interface that gets context from some source and puts it in a map
+ * for a reranking processor to use
+ */
+public interface ContextSourceFetcher {
+
+    /**
+     * Fetch the information needed in order to rerank.
+     * That could be as simple as grabbing a field from the search request or
+     * as complicated as a lookup to some external service
+     * @param searchRequest the search query
+     * @param searchResponse the search results, in case they're relevant
+     * @param listener be async
+     */
+    public void fetchContext(SearchRequest searchRequest, SearchResponse searchResponse, ActionListener<Map<String, Object>> listener);
```

Comment:
nit: `public` is redundant for interface methods

Reply:
ok